tests/prompt_strategies/test_chat_templates_advanced.py

```diff
@@ -912,5 +912,84 @@ def verify_labels(labels_span, should_train, context_message):
         LOG.debug(f"Final input_ids: {input_ids}")
 
 
+class TestOptionalMessageFields:
+    """Test class for optional message fields functionality in chat templates."""
+
+    def test_combined_optional_fields_with_template(self, request):
+        """Test both default and custom optional fields with template rendering."""
+        LOG.info("Testing combined optional fields with template")
+
+        # Template that includes both tool_calls and thoughts
+        template = """{% for message in messages %}
+{{'<|im_start|>'}}{{ message['role'] }}
+{% if message['thoughts'] is defined %}[Thoughts: {{ message['thoughts'] }}]{% endif %}
+{{ message['content'] }}{{'<|im_end|>'}}
+{% endfor %}"""
+
+        tokenizer = deepcopy(request.getfixturevalue("llama3_tokenizer"))
+
+        test_data = {
+            "messages": [
+                {"role": "system", "content": "You are an AI assistant."},
+                {"role": "user", "content": "What is the temperature in Paris?"},
+                {
+                    "role": "assistant",
+                    "content": "Let me help with that calculation",
+                    "thoughts": "We should take care to convert the temperature to Fahrenheit",
+                },
+            ]
+        }
+
+        strategy = ChatTemplateStrategy(
+            ChatTemplatePrompter(
+                tokenizer,
+                chat_template=template,
+                message_field_role="role",
+                message_field_content="content",
+                optional_message_fields=["thoughts"],
+            ),
+            tokenizer=tokenizer,
+            train_on_inputs=False,
+            sequence_len=512,
+            roles_to_train=["assistant"],
+        )
+
+        res = strategy.tokenize_prompt(test_data)
+        turns = strategy.get_conversation_thread(test_data)
+        labels = res["labels"]
+        input_ids = res["input_ids"]
+
+        LOG.info(f"Labels: {labels}")
+        LOG.info(f"Input IDs: {input_ids}")
```

Comment:
```suggestion
        LOG.debug(f"Labels: {labels}")
        LOG.debug(f"Input IDs: {input_ids}")
```
log probably not needed in the tests, but can convert these to debug if we want to keep them.